tests/model_registry/conftest.py

```diff
@@ -60,121 +61,190 @@ def model_registry_namespace(updated_dsc_component_state_scope_class: DataScienc
 
 @pytest.fixture(scope="class")
 def model_registry_db_service(
-    admin_client: DynamicClient, model_registry_namespace: str, is_model_registry_oauth: bool
+    pytestconfig: Config,
+    admin_client: DynamicClient,
+    model_registry_namespace: str,
+    teardown_resources: bool,
 ) -> Generator[Service, Any, Any]:
-    with Service(
-        client=admin_client,
-        name=DB_RESOURCES_NAME,
-        namespace=model_registry_namespace,
-        ports=[
-            {
-                "name": "mysql",
-                "nodePort": 0,
-                "port": 3306,
-                "protocol": "TCP",
-                "appProtocol": "tcp",
-                "targetPort": 3306,
-            }
-        ],
-        selector={
-            "name": DB_RESOURCES_NAME,
-        },
-        label=get_model_registry_db_label_dict(db_resource_name=DB_RESOURCES_NAME),
-        annotations={
-            "template.openshift.io/expose-uri": r"mysql://{.spec.clusterIP}:{.spec.ports[?(.name==\mysql\)].port}",
-        },
-    ) as mr_db_service:
+    if pytestconfig.option.post_upgrade:
+        mr_db_service = next(
+            Service.get(client=admin_client, name=DB_RESOURCES_NAME, namespace=model_registry_namespace), None
```

Comment:
You already know the name and namespace. You don't need get() call. Instead, you can make call like this 
```
yield Secret(name=name=DB_RESOURCES_NAME, namespace=model_registry_namespace, ensure_exists=True)

Reply:
I had tried doing this, but then the yielded resource would create problems for other fixtures/function (e.g. in the mysql config we would be unable to get the Secret data (database/username)) - we would need to return `resource.instance`, or call `.instance` on the resource in the other code, but this would not work for the usual resource yielded by the fixture when we are not in post upgrade. I've updated at least one fixture that had problems with this to use values from constants instead, so maybe it's worth another try. I'll let you know how it goes.

Reply:
Ok, tried again, I think with the changes I had made to the mysql fixture we are good with yielding the resources like this as I didn't encounter any failures in pre/post upgrade. Let's keep an eye on this if it were to ever fail.